### ISSUE TITLE
The family parameter in gnt-network is unclear

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#424]
Originally reported of Google Code with ID 368.

```
It seems the family parameter might be a leftover from the past: is it the case? Networks seem to have both a subnet and subnet6, and there seems to be no way to specify the family, nor any mention on it in the manpage. What is that about?

thanks,

Guido

```
Originally added on 2013-02-07 17:30:48 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Well currently we have IPv4 pool that can be associated with an IPv6 subnet. The NIC
can obtain IPv6 address derived from its MAC and the IPv6 subnet prefix via mac2eui64.

In the future we could have strict IPv6 Networks without the need of IPv4 pool. To this end the family is needed to distinguish each case.

The plan can be to have a network object that wraps IPv4 and IPv6 networks. IPv4 networks will still need AddressPool mechanism in network.py but IPv6 ones will not
(impossible and meaningless to support it).

Do you agree with all that?

dimara
```
Originally added on 2013-02-08 16:17:58 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Agreed about supporting either IPv4, IPv6 (without a pool), or both.

Still, do we need the family (and size) parameters? it seems if "network" is set the network has ipv4, if "network6" is set it has ipv6, if both are set we support both, no? :)

Thanks,

Guido

```
Originally added on 2013-02-08 16:34:22 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Well size can go away for sure. It can derive from the subnet via ipaddr module.

Currently Ganeti networks are strictly associated with IPv4 pools. You cannot create
a network without passing an IPv4 subnet. Associating an IPv6 subnet to a Ganeti network is optional. A NIC can obtain an IP from the IPv4 pool and this value is
stored in ip slot inside the NIC config object.

As far as the family is concerned and having in mind all the above, I mention the following scenario/design:

Remove family from the Network object.
Make --network option *not* mandatory.
A user will be able create a network without passing a subnet (v4 or v6). This way,
mac prefix, tags, connectivity mode and link will refer to a collision domain (Ethernet network).
If an --network is passed then create an IP pool and support the current functionality (ip=pool, GenerateIP(), etc.).
If an --network6 is passed then no pool will be created.
A NIC can have both IPv4 and IPv6 addresses so an extra ip6 slot in NIC config
object is needed. This will allow us define something like ip6=generate and get
a valid IPv6 inside the corresponding subnet (via mac + subnet + mac2eui64).

I guess removing family and size can go now to devel-2.7 and the rest will be applied on top of master.  

Any comments on the above? Should I proceed with patches?

Thanks,
dimara


```
Originally added on 2013-02-08 20:17:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```





Hi,

Yes, considering that currently ganeti networks are strictly IPv4 both family and size can go (they are sort of obvious) :) So let's make them go (for devel-2.7).

Agreed on all the rest of the plan *except* the ip6 slot: we should just convert the ip slot to be a list of IPs, either version 4 or 6 (or if that's problematic a list of IP objects with a family and addr slots). This way we can also support multiple IPv4 or IPv6 per nic.

Thanks!

Guido


```
Originally added on 2013-02-09 05:58:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-02-11 15:32:45 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
